src/ecmult_gen_impl.h

```diff
@@ -225,13 +225,14 @@ static void secp256k1_ecmult_gen(const secp256k1_ecmult_gen_context *ctx, secp25
         bit_pos = comb_off;
         for (block = 0; block < COMB_BLOCKS; ++block) {
 #if COMB_GROUPED
-            bits = (recoded[bit_pos >> 5] >> (bit_pos & 0x1F)) & ((1 << COMB_TEETH) - 1);
-            bit_pos += COMB_SPACING * COMB_TEETH;
+            bits = recoded[bit_pos >> 5] >> (bit_pos & 0x1F);
+            bit_pos += COMB_TEETH;
 #else
             bits = 0;
             for (tooth = 0; tooth < COMB_TEETH; ++tooth) {
-                bit = (recoded[bit_pos >> 5] >> (bit_pos & 0x1F)) & 1;
-                bits |= bit << tooth;
+                bit = recoded[bit_pos >> 5] >> (bit_pos & 0x1F);
+                bits &= ~(1 << tooth);
```

Comment:
I think this bit-clearing line isn't actually needed, because `bits` starts at 0 and `1 << tooth` is unique on each iteration.

Reply:
Note that ‘bit’ may now contain junk (or noise) bits beyond bit 0. Iterations after the first therefore need to clear the target bit (and we try to leave noise in the high bits). The change is in response to [1], which I think @gmaxwell mentioned on IRC a while back.

[1] https://www.usenix.org/system/files/conference/usenixsecurity18/sec18-alam.pdf

Reply:
Ah! I understand, I was misreading `bit << tooth` as `(bit & 1) << tooth`.

Reply:
Because the variable is named `bit` :P